datafusion/core/src/config.rs

```diff
@@ -134,9 +135,9 @@ pub const OPT_HASH_JOIN_SINGLE_PARTITION_THRESHOLD: &str =
 /// Definition of a configuration option
 pub struct ConfigDefinition {
     /// key used to identifier this configuration option
-    key: String,
+    key: Cow<'static, str>,
```

Comment:
What would you think about using `Arc` here instead?

`Arc` might would be more consistent  with the rest of the DataFusion codebase as well as help systems like Ballista where the configuration was serialized / deserialized. 

I believe with a `Cow` and without any extra work, the deserialized `ConfigOptions` will still be owned (and copied) Strings

Reply:
Hmm... You raise a good point r.e. serialization, perhaps we should just make ConfigOptions a struct... I'll leave this open if anyone else has any thoughts